auth/handler.go

```diff
@@ -0,0 +1,64 @@
+package auth
+
+import (
+	"net/http"
+
+	"github.com/ONSdigital/dp-permissions/permissions"
+	"github.com/ONSdigital/go-ns/common"
+	"github.com/ONSdigital/log.go/log"
+)
+
+//go:generate moq -out authtest/auth_mocks.go -pkg authtest . PermissionAuthenticator
+
+const (
+	CollectionIDHeader = "Collection-Id"
+	DatasetIDParam     = "dataset_id"
```

Comment:
The setting of this has interesting implications now this is a separate library :thinking: It would mean serivces which define this string differently (say 'dsid' or something) would have to change their route declarations to use the value hardcoded in this library. I dont know if it would be sensible to pass this variable in, or set it on configure?

i think we have fairly standard routing so im not sure its a problem we'll actually face, and not the end of the world if we dont solve it, its just a coupling between the library and the services thats worth being aware of